synapse/handlers/profile.py

```diff
@@ -307,7 +307,11 @@ async def check_avatar_size_and_mime_type(self, mxc: str) -> bool:
         if not self.max_avatar_size and not self.allowed_avatar_mimetypes:
             return True
 
-        server_name, _, media_id = parse_and_validate_mxc_uri(mxc)
+        host, port, media_id = parse_and_validate_mxc_uri(mxc)
+        if port is not None:
+            server_name = host + ":" + str(port)
```

Comment:
As an aside: I'm really rather surprised that we allow setting the avatar to a remote server. What is your usecase for this?

Reply:
Well, I am just running synapse locally on 8080 port and trying to set an avatar that I uploaded (so have a mxc url for it), but found this function was treating that as a remote server since it didn't mind the non-default port in use.

Reply:
oh yes, of course